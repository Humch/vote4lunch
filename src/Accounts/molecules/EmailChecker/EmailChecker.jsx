import React from 'react';
import { connect } from 'react-redux';

import PropTypes from 'prop-types';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSpinner } from '@fortawesome/free-solid-svg-icons';

import { ValueChecker } from '../../../Core';

const EmailChecker = ({ payload, loading }) => {
  let rendering = '';
  if (loading) {
    rendering = (
      <FontAwesomeIcon
        icon={faSpinner}
        spin
      />
    );
  } else if (!payload.exist) {
    rendering = <ValueChecker check />;
  } else {
    rendering = <ValueChecker check={false} />;
  }
  return (
    <span>
      { rendering }
    </span>
  );
};

EmailChecker.propTypes = {
  payload: PropTypes.shape({
    exist: PropTypes.bool.isRequired
  }),
  loading: PropTypes.bool.isRequired,
};

const mstp = ({ emailChecker }) => ({
  payload: emailChecker.payload,
  loading: emailChecker.loading,
});

export default connect(mstp)(EmailChecker);
