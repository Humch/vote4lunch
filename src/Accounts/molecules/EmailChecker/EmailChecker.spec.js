/* eslint-disable no-undef */
import React from 'react';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import EmailChecker from './EmailChecker';

describe('EmailChecker Snapshot', () => {
  const initialState = {
    emailChecker: {
      payload: { exist: false },
      loading: false,
    }
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of EmailChecker', () => {
    const renderedValue = renderer.create(<EmailChecker store={store} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('EmailChecker Snapshot', () => {
  const initialState = {
    emailChecker: {
      payload: { exist: false },
      loading: true,
    }
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of EmailChecker with loading', () => {
    const renderedValue = renderer.create(<EmailChecker store={store} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('EmailChecker Snapshot', () => {
  const initialState = {
    emailChecker: {
      payload: { exist: true },
      loading: false,
    }
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of EmailChecker with exist true', () => {
    const renderedValue = renderer.create(<EmailChecker store={store} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
