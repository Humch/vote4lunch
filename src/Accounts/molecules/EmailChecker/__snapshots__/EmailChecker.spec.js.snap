// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailChecker Snapshot capturing Snapshot of EmailChecker 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-check fa-w-16 "
  color="green"
  data-icon="check"
  data-prefix="fas"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`EmailChecker Snapshot capturing Snapshot of EmailChecker with exist true 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-ban fa-w-16 "
  color="red"
  data-icon="ban"
  data-prefix="fas"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M256 8C119.034 8 8 119.033 8 256s111.034 248 248 248 248-111.034 248-248S392.967 8 256 8zm130.108 117.892c65.448 65.448 70 165.481 20.677 235.637L150.47 105.216c70.204-49.356 170.226-44.735 235.638 20.676zM125.892 386.108c-65.448-65.448-70-165.481-20.677-235.637L361.53 406.784c-70.203 49.356-170.226 44.736-235.638-20.676z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`EmailChecker Snapshot capturing Snapshot of EmailChecker with loading 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-spinner fa-w-16 fa-spin "
  data-icon="spinner"
  data-prefix="fas"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;
